app/models/grda_warehouse/tasks/identify_duplicates.rb

```diff
@@ -44,55 +45,67 @@ def identify_duplicates
       # If we do find a match:
       #   create the associated WarehouseClient
 
-      Rails.logger.info "Matching #{unprocessed.count} unprocessed clients"
+      unprocessed_count = unprocessed.count
+      Rails.logger.info 'Matching #unprocessed_count} unprocessed clients'
       matched = 0
       new_created = 0
+
+      # Find all potential matches
+      more_than_one_match = find_merge_candidates_for_unprocessed
+      found_ids = more_than_one_match.flatten.uniq.to_set
+
+      # At this point more_than_one_match contains pairs of destination client id and source client id where the
+      # source client should be merged into the destination client.
+      # If the source client does not exist in any of the pairs, a new destination client should be created for it.
+      destination_clients_by_id = GrdaWarehouse::Hud::Client.where(id: found_ids.to_a).index_by(&:id)
```

Comment:
restrict for extra safety?
```suggestion
      destination_clients_by_id = GrdaWarehouse::Hud::Client.destination.where(id: found_ids.to_a).index_by(&:id)
```

Reply:
That scope seems to break the tests.